exporter/clickhousemetricsexporter/exporter.go

```diff
@@ -76,13 +76,15 @@ func NewPrwExporter(cfg *Config, set exporter.CreateSettings) (*PrwExporter, err
 	if err != nil {
 		return nil, errors.New("invalid endpoint")
 	}
+	maxIdleConnections := cfg.RemoteWriteQueue.NumConsumers + 1
 
 	userAgentHeader := fmt.Sprintf("%s/%s", strings.ReplaceAll(strings.ToLower(set.BuildInfo.Description), " ", "-"), set.BuildInfo.Version)
 
 	params := &ClickHouseParams{
 		DSN:                  cfg.HTTPClientSettings.Endpoint,
 		DropDatabase:         false,
-		MaxOpenConns:         75,
+		MaxIdleConns:         maxIdleConnections,
+		MaxOpenConns:         maxIdleConnections + 10,
```

Comment:
what is the reasoning for this logic?

Reply:
There are 6-7 PrepareBatch writes in the metrics exporter. I added a few more just in case.

Reply:
`WithConnectionRelease` should make use of 1 connection for all these 6-7 batches? https://github.com/ClickHouse/clickhouse-go/blob/eb76971a3050d2c83ae5790162e0d52ba87712f8/README.md#preparebatch-options

Reply:
I think you are correct. I initially thought wouldn't be immediately released so it might cause a problem with the next preparebatch. Let me go through the changeset once and check.

Reply:
Concurrent connections to clickhouse is going to be pain sometime soon. We should keep a watch.